subprojects/platform-native/src/testFixtures/groovy/org/gradle/nativeplatform/fixtures/app/IncrementalElement.java

```diff
@@ -0,0 +1,250 @@
+                TestFile file = projectDir.file(beforeFile.withPath("src/" + sourceSetName));
+                file.assertExists();
+
+                file.write(afterFile.getContent());
+            }
+
+            @Override
+            public void populateBefore(List<SourceFile> beforeSourceFiles) {
+                beforeSourceFiles.add(beforeFile);
+            }
+
+            @Override
+            public void populateAfter(List<SourceFile> afterSourceFiles) {
+                afterSourceFiles.add(afterFile);
+            }
+        };
+    }
+
+    /**
+     * Returns a transform that delete the before element.
+     */
+    protected Transform delete(SourceFileElement beforeElement) {
+        assert beforeElement.getFiles().size() == 1;
+        final String sourceSetName = beforeElement.getSourceSetName();
+        final SourceFile beforeFile = beforeElement.getSourceFile();
+
+        return new Transform() {
+            @Override
+            public void applyChangesToProject(TestFile projectDir) {
+                TestFile file = projectDir.file(beforeFile.withPath("src/" + sourceSetName));
+                file.assertExists();
+
+                file.delete();
+            }
+
+            @Override
+            public void populateBefore(List<SourceFile> beforeSourceFiles) {
+                beforeSourceFiles.add(beforeFile);
+            }
+
+            @Override
+            public void populateAfter(List<SourceFile> afterSourceFiles) {}
+        };
+    }
+
+    /**
+     * Returns a transform that add the after element.
+     */
+    protected Transform add(SourceFileElement afterElement) {
+        assert afterElement.getFiles().size() == 1;
+        final String sourceSetName = afterElement.getSourceSetName();
+        final SourceFile afterFile = afterElement.getSourceFile();
+
+
+        return new Transform() {
+            @Override
+            public void applyChangesToProject(TestFile projectDir) {
+                TestFile file = projectDir.file(afterFile.withPath("src/" + sourceSetName));
+
+                file.assertDoesNotExist();
+
+                afterFile.writeToDir(projectDir);
+            }
+
+            @Override
+            public void populateBefore(List<SourceFile> beforeSourceFiles) {}
+
+            @Override
+            public void populateAfter(List<SourceFile> afterSourceFiles) {
+                afterSourceFiles.add(afterFile);
+            }
+        };
+    }
+
+    /**
+     * Returns a transform that rename the before element to {@code renamed-} followed by the original name.
+     */
+    protected Transform rename(SourceFileElement beforeElement) {
+        assert beforeElement.getFiles().size() == 1;
+        final String sourceSetName = beforeElement.getSourceSetName();
+        final SourceFile beforeFile = beforeElement.getSourceFile();
+        final SourceFile afterFile = new SourceFile(beforeFile.getPath(), "renamed-" + beforeFile.getName(), beforeFile.getContent());
```

Comment:
I wonder if this method should take a `String` argument that is the renamed file name.  You might have an overload that only takes the `SourceFileElement` and defaults to "renamed-xxx".  Not sure if there might be a scenario where the renamed file name is important or not.  Or you could just leave this as is - easy enough to add that functionality if it's ever required - your call.

Reply:
I added an overload to choose the rename prefix.